feature-gate/program/src/feature_id.rs

```diff
@@ -0,0 +1,37 @@
+//! Module for managing feature IDs
+
+// use {
+//     crate::error::FeatureGateError,
+//     solana_program::{program_error::ProgramError, pubkey::Pubkey},
+//     solana_sdk::feature_set::FEATURE_NAMES,
+// };
+
+// /// Returns `true` if the feature ID exists in the current feature set.
+// fn feature_exists(feature_id: &Pubkey) -> bool {
+//     FEATURE_NAMES.iter().any(|(id, _)| id == feature_id)
+// }
+
+// /// Derives the feature ID from a authority's address.
+// pub fn derive_feature_id(authority: &Pubkey) -> Result<Pubkey, ProgramError> {
+//     let nonce = 0u16; // u16::MAX = 65_535 features!
+//     let mut feature_id = Pubkey::find_program_address(&[b"feature"], &crate::id()).0;
+//     while feature_exists(&feature_id) {
+//         nonce.checked_add(1).ok_or(FeatureGateError::Overflow)?;
+//         feature_id = Pubkey::find_program_address(
+//             &[b"feature", &nonce.to_le_bytes(), &authority.to_bytes()],
+//             &crate::id(),
+//         )
+//         .0;
+//     }
+//     Ok(feature_id)
+// }
```

Comment:
I _really_ wish this would work, but a program can't import solana_sdk :(

Any other ideas @joncinque  ?

Reply:
It also won't work because once an eng calls `derive_feature_id()` once and puts an entry in FEATURE_NAMES to start using the gate for runtime changes, `derive_feature_id()` won't return that same key again for the actual activation.
Probably have to let authorities handle their own noncing or seed determination in some fashion...?